diskmaker/discovery/discovery.go

```diff
@@ -182,8 +182,12 @@ func getDiscoverdDevices(blockDevices []internal.BlockDevice) []v1alpha1.Discove
 			klog.Warningf("failed to parse size for the device %q. Error %v", blockDevice.Name, err)
 		}
 
+		path, err := blockDevice.GetDevPath()
+		if err != nil {
+			klog.Warningf("failed to parse path for the device %q. Error %v", blockDevice.KName, err)
+		}
 		discoveredDevice := v1alpha1.DiscoveredDevice{
-			Path:     fmt.Sprintf("/dev/%s", blockDevice.Name),
+			Path:     path,
```

Comment:
Can we just get away by changing this line to:

```
Path: fmt.Sprintf("/dev/%s", blockDevice.KName)
```

I am not sure why we need device mapper path from `/dev/mapper` for any of the stuff.

Reply:
@gnufied well, we probably don't want to report invalid path in discovery result. That's the only reason I believe. Without the patch you'll get `/dev/mpath` or something like that - such path will never exist for multipath.

Reply:
No, I meant if you used `KName` here, it will point to a valid device. Output of `lsblk` with `KNAME` always points to a valid device.  Having said that, there are pros and cons of both the approaches:

Pro of using `KName` here is - we can use single device mechanism for every device type. 

Pro of using mpath name is, users can easily tell system has selected an mpath device. But then how we show different devices becomes inconsistent. I also think that, mpath device names btw changes between reboots (so can `KName` to be fair), so in the end all of the code must use `dev/disk/by-id` path anyways. But symlink in `/dev/disk/by-id` always points to `KName` device in `/dev/<kname>` location.

Reply:
> No, I meant if you used `KName` here, it will point to a valid device. Output of `lsblk` with `KNAME` always points to a valid device. Having said that, there are pros and cons of both the approaches:

Ah, I see. I misread that and though you're suggesting to use what the original code was using, so `blockDevice.Name` for the path.

Those are valid points, the reason I went with `/dev/mapper` was that the original code used `Name` instead of `KName` and I assumed we'd like to keep this naming - volume discovery result looked like this:

```
  - deviceID: /dev/disk/by-id/dm-name-mpatha
    fstype: ""
    model: ""
    path: /dev/mpatha
```

> Pro of using `KName` here is - we can use single device mechanism for every device type.
>
> Pro of using mpath name is, users can easily tell system has selected an mpath device. But then how we show different devices becomes inconsistent. I also think that, mpath device names btw changes between reboots (so can `KName` to be fair), so in the end all of the code must use `dev/disk/by-id` path anyways. But symlink in `/dev/disk/by-id` always points to `KName` device in `/dev/<kname>` location.

Not sure what you mean by "single device mechanism" but I assume it's the way we parse device paths in the code. So with `KName` approach the advantage would be simpler code for sure.

As for the inconsistency in showing the devices: without further refactoring the discovery is not optimal, regardless of which approach of showing path we choose. See the `LocalVolumeDiscoveryResult` I've posted above - we show all underlying devices of mpath (`type: disk`) along with multipath device (`type: mpath`) which is duplicated for every underlying device. Ideally it should show only one mpath? That means adding code that will reliably verify what mpath devices lead to exactly - wondering if it's even worth dealing with.